internal/rbd/nodeserver.go

```diff
@@ -154,6 +154,7 @@ func populateRbdVol(
 	volID := req.GetVolumeId()
 	isBlock := req.GetVolumeCapability().GetBlock() != nil
 	disableInUseChecks := false
+	rv := &rbdVolume{}
```

Comment:
Why create an empty `rbdVolume`, and not assign `nil` to it? This changes the return values in error cases. Returning `nil` has my preference, as using the volume after an error is prevented that way. (Defined behaviour, panic, instead of undefined behaviour with empty values.)

Reply:
This is to the proper cleanup of connection we are already doing it in other places also https://github.com/ceph/ceph-csi/blob/eb40fbcb1855f3a6812d71b9cce39e5b0f12cd87/internal/rbd/rbd_util.go#L1048 if we don't want to do it, i need to handle Destroy in defer in the called function.